### PR TITLE
Introduce capability struct with provider specific template context schema

### PR DESCRIPTION
**Pull request by pranavgaikwad**
Addresses https://github.com/konveyor/analyzer-lsp/issues/24

~openapi-go does not support v3, I used it as it seemed sufficient for our use-case. Open to discussing other options~

Updated to use kin-openapi/openapi3

### CODE REVIEW

**Review by fabianvf — COMMENTED** [review]
looks good to me, do we want to add a way to print the templatecontext schemas as JSON for users to introspect here or do that in a follow-up?

**Comment by pranavgaikwad** [comment]
@fabianvf openapi3.Schema implements MarshalJSON(), do we want to explicitely add something?

**Comment by fabianvf** [comment]
@pranavgaikwad yeah we don't need to reimplement that, I mean more from the CLI side something to display this output

File: provider/provider.go [hunk 1]
```diff
@@ -19,6 +20,8 @@ import (
 type Client interface {
 	Capabilities() ([]string, error)
 
+	TemplateContext() (openapi3.Schema, error)
```
Comment (shawn-hurley) [inline provider/provider.go:23]: I wonder if this should be a part of the Capabilities return and if we just make that struct. I could also see it being helpful to have a context per capability.

Comment (fabianvf) [reply provider/provider.go:23]: I like that idea

**Comment by pranavgaikwad** [comment]
@fabianvf I will address the CLI thing in a follow-up. I think the CLI deserves its own separate issue.

File: provider/lib/lib.go [hunk 2]
```diff
@@ -13,6 +14,11 @@ var builtinConfig = Config{
 	Location: ".",
 }
 
+type Capability struct {
+	Capability      string
```
Comment (shawn-hurley) [inline provider/lib/lib.go:18]: Capability.Capability stutters I think, I think that we maybe want it to be the Name of the cap?

**Review by shawn-hurley — APPROVED** [review]
One naming concern but other than that looks good to me.

**Pull request merged**